qiskit/algorithms/minimum_eigensolvers/adapt_vqe.py

```diff
@@ -0,0 +1,311 @@
+
+        Args:
+            indices: The list of chosen gradient indices.
+
+        Returns:
+            Whether repeating sequences of indices have been detected.
+        """
+        cycle_regex = re.compile(r"(\b.+ .+\b)( \b\1\b)+")
+        # reg-ex explanation:
+        # 1. (\b.+ .+\b) will match at least two numbers and try to match as many as possible. The
+        #    word boundaries in the beginning and end ensure that now numbers are split into digits.
+        # 2. the match of this part is placed into capture group 1
+        # 3. ( \b\1\b)+ will match a space followed by the contents of capture group 1 (again
+        #    delimited by word boundaries to avoid separation into digits).
+        # -> this results in any sequence of at least two numbers being detected
+        match = cycle_regex.search(" ".join(map(str, indices)))
+        logger.debug("Cycle detected: %s", match)
+        logger.info("Alternating sequence found. Finishing.")
+        # Additionally we also need to check whether the last two numbers are identical, because the
+        # reg-ex above will only find cycles of at least two consecutive numbers.
+        # It is sufficient to assert that the last two numbers are different due to the iterative
+        # nature of the algorithm.
+        return match is not None or (len(indices) > 1 and indices[-2] == indices[-1])
+
+    def compute_minimum_eigenvalue(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+    ) -> AdaptVQEResult:
+        """Computes the minimum eigenvalue.
+
+        Args:
+            operator: Operator whose minimum eigenvalue we want to find.
+            aux_operators: Additional auxiliary operators to evaluate.
+
+        Raises:
+            QiskitError: If a solver other than :class:`~VQE` or an ansatz other than
+                :class:`~.EvolvedOperatorAnsatz` is provided or if the algorithm finishes due
+                to an unforeseen reason.
+            AlgorithmError: If `quantum_instance` is not provided.
+            QiskitError: If the chosen gradient method appears to result in all-zero gradients.
+
+        Returns:
+            An :class:`~.AdaptVQEResult` which is a :class:`~.VQEResult` but also but also
+            includes runtime information about the AdaptVQE algorithm like the number of iterations,
+            termination criterion, and the final maximum gradient.
+        """
+        if not isinstance(self._tmp_ansatz, EvolvedOperatorAnsatz):
+            raise QiskitError("The AdaptVQE ansatz must be of the EvolvedOperatorAnsatz type.")
+
+        # Overwrite the solver's ansatz with the initial state
+        solver_ansatz = copy.deepcopy(self.solver.ansatz)
+        self._excitation_pool = copy.deepcopy(solver_ansatz.operators)
+        self.solver.ansatz = self._tmp_ansatz.initial_state
```

Comment:
VQE has a check that the anstaz is parameterized. By the time VQE is called to compute the min eig value has this already been replaced

Reply:
This checks happens at the beginning of `VQE.compute_minimum_eigenvalue`. This method has not been called yet, here, but also will only get called, after a parameterized excitation has been re-added to the solvers' ansatz at the end of the first `while` loop iteration.